warp10/src/main/java/io/warp10/script/functions/MACROMAPPER.java

```diff
@@ -38,7 +43,7 @@
 
 public class MACROMAPPER extends NamedWarpScriptFunction implements WarpScriptStackFunction {
 
-  public static class MacroMapperWrapper extends NamedWarpScriptFunction implements WarpScriptMapperFunction, WarpScriptReducerFunction, WarpScriptBucketizerFunction, WarpScriptAggregatorOnListsFunction {
+  public static class MacroMapperWrapper extends NamedWarpScriptFunction implements WarpScriptMapperFunction, WarpScriptReducerFunction, WarpScriptBucketizerFunction, WarpScriptMapper, WarpScriptReducer, WarpScriptBucketizer, WarpScriptAggregatorRemoveNulls, WarpScriptAggregatorKeepNulls, WarpScriptAggregatorSkipIfAnyNull, WarpScriptAggregatorFailIfAnyNull {
```

Comment:
The decorator interface has a default method that the decorator applies. A decorated aggregator that extends this interface can override this default method.

Reply:
To make this bearable in the long run, we can move the default method inside the decorator, and make the decorator interface optional, so that only aggregators that need to override the default behavior extend it.